src/main/java/net/minecraftforge/client/event/RenderWorldPreTranslucentEvent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+
+import net.minecraft.client.renderer.IRenderTypeBuffer.Impl;
+import net.minecraft.client.renderer.Matrix4f;
+import net.minecraft.client.renderer.WorldRenderer;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * This event is called immediately before translucent blocks are rendered in the WorldRenderer class.
+ * 
+ * It is useful for rendering in a similar manner to RenderWorldLastEvent without the result being
+ * invisible behind translucent blocks such as stained glass or water.
+ * However, due to the nature of translucent block rendering, translucent blocks in the world
+ * will be invisible behind anything translucent rendered from this event.
+ */
+
+public class RenderWorldPreTranslucentEvent extends Event
+{
+    private final WorldRenderer context;
+    private final MatrixStack mat;
+    private final Impl buffers;
+    private final float partialTicks;
+    private final Matrix4f projectionMatrix;
+    private final long finishTimeNano;
+
+    public RenderWorldPreTranslucentEvent(WorldRenderer context, MatrixStack mat, Impl buffers, float partialTicks, Matrix4f projectionMatrix, long finishTimeNano)
+    {
+        this.context = context;
+        this.mat = mat;
+        this.buffers = buffers;
+        this.partialTicks = partialTicks;
+        this.projectionMatrix = projectionMatrix;
+        this.finishTimeNano = finishTimeNano;
+    }
+
+    public WorldRenderer getContext()
+    {
+        return context;
+    }
+
+    public MatrixStack getMatrixStack()
+    {
+        return mat;
+    }
+    
+    public Impl getRenderTypeBuffers()
```

Comment:
I'd use `IRenderTypeBuffer.Impl` since this is an inner class reference. (same with field and parameter declarations)